README


This archive contains the distribution Dist-Zilla-Plugin-Run,
version 0.029:

  Run external commands and code at specific phases of Dist::Zilla

This software is copyright (c) 2010 by L<Raudssus Social Software|https://raudss.us/>.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v5.029.



Changes
Revision history for Dist-Zilla-Plugin-Run

0.029     2015-01-05 01:59:05Z
  - added new "fatal_errors" option (defaults to true, as always)

0.028     2014-11-24 02:35:12Z
  - now properly evaluating formatting codes inside 'eval' lines

0.027     2014-11-12 01:41:50Z
  - new formatter code added: "%t", for indicating a trial release (PR#9,
    Thomas Sibley)

0.026     2014-09-06 06:42:06Z
  - all test reports are green! release the hounds!

0.025     2014-08-18 03:36:32Z (TRIAL RELEASE)
  - fix use of path separator in test to avoid more failures on MSWin32
  - fix use of :crlf mode in test to avoid yet more failures on MSWin32

0.024     2014-08-16 00:36:01Z (TRIAL RELEASE)
  - fix use of path separator in test to avoid failure on MSWin32
  - new "eval" option, for executing Perl code

0.023     2014-08-06 05:33:20Z
  - convert uses of Path::Class to Path::Tiny
  - dump config into metadata when [MetaConfig] used, but the
    'censor_commands' option can be used to avoid this

0.022     2014-07-24 15:36:32Z
  - more complete metadata and author tests

0.021     2014-02-23 23:31:47Z
  - Lazy load runtime dependencies. Thanks Olivier Mengué!
  - unroll author pluginbundle, to remove some unneeded and broken
    prerequisites

0.020     2013-04-21 00:02:21Z
  - Switching to [@Author::GETTY] dzil bundle
  - Adding .travis.yml for travis-ci

0.019     2013-04-15 14:52:25Z
  - Delay evaluation of some vars (like dist version) until needed.
    Thanks, Tatsuhiko Miyagawa!

0.018     2013-03-02 17:46:39Z
  - Added new options, cleaned up documentation

0.017     2013-02-17 04:20:54Z
  - Fix silly bug in tests for non-unix platforms

0.016     2013-02-13 14:49:47Z
  - Add 'run_no_trial' option for running commands
    only when it is not a trial build/release.

0.015     2013-02-04 00:25:50Z
  - Copy examples for all runners to Synposis on main page

0.014     2013-01-19 03:21:24Z
  - Add Run::AfterMint plugin for running commands after making a new dist.

0.013     2011-11-09 13:23:08Z
  - Fix Pod spelling error pointed out by Debian Perl Group (rt-72278).

0.012     2011-11-06 00:30:03Z
  - Add '%x' for perl path (similar to $^X)
  - Use strict and warnings in each module
  - Test fixes

0.011     2011-10-28 15:25:04Z
  - Explicitly stringify '%a' to keep String::Formatter happy.
    Make sure it's tested.
    Closes rt-72008. Thanks to Olivier Mengué (dolmen) for reporting.

0.010     2011-09-10 18:06:33Z
  - Define '%p' as the OS path separator in case anybody ever needs
    cross-platform dist.ini commands
  - Enable the tests for the Run::Test plugin
  - Make tests cross-platform

0.009     2011-09-07 14:34:14Z
  - added the [Run::Test] plugin

0.008     2011-08-09 16:31:33Z
  - Stream command output incrementally instead of waiting
    until the command is done. (RWSTAUNER)
    Closes gh-4.  Thanks to SamuraiJack for the suggestion.

  - Show correct exit status on command failure.

0.007     2011-08-08 03:30:39Z
  - Define other (constant) string format conversions (%n, %v);
    Retain positional %s for backward compatibility. (RWSTAUNER)

0.006     2011-05-21 00:58:17Z
    - Fixed wrong bangpath

0.005     2011-04-29 20:54:18Z

0.005     2011-01-21 10:21:08Z

0.004     2010-12-14 20:19:54Z
    - Minimal pod changes

0.003     2010-12-14 02:18:00Z
    - Complete Change of API

0.002     2010-12-13 17:07:27Z

0.001     2010-12-12 03:14:54Z
    - First release


Makefile.PL
# This Makefile.PL for Dist-Zilla-Plugin-Run was generated by
# Dist::Zilla::Plugin::MakeMaker::Fallback 0.016
# and Dist::Zilla::Plugin::MakeMaker::Awesome 0.32.
# Don't edit it but the dist.ini and plugins used to construct it.

use strict;
use warnings;

BEGIN {
my %configure_requires = (
    'Module::Build::Tiny' => '0.039',
);

my @missing = grep {
    ! eval "require $_; $_->VERSION($configure_requires{$_}); 1"
} keys %configure_requires;

if (not @missing)
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";
}
else
{
    $ENV{PERL_MM_FALLBACK_SILENCE_WARNING} or warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

If you're installing manually, please retrain your fingers to run Build.PL
when present instead.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are doing something overly clever, in which case you
should consider setting the 'prefer_installer' config option in CPAN.pm, or
'prefer_makefile' in CPANPLUS, to 'mb" and '0' respectively.

You can also silence this warning for future installations by setting the
PERL_MM_FALLBACK_SILENCE_WARNING environment variable.

EOW

    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));
}
}

use 5.006;
use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Run external commands and code at specific phases of Dist::Zilla",
  "AUTHOR" => "Torsten Raudssus <torsten\@raudss.us> L<https://raudss.us/>",
  "CONFIGURE_REQUIRES" => {
    "Module::Build::Tiny" => "0.039"
  },
  "DISTNAME" => "Dist-Zilla-Plugin-Run",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "Dist::Zilla::Plugin::Run",
  "PREREQ_PM" => {
    "Dist::Zilla::Role::AfterBuild" => 0,
    "Dist::Zilla::Role::AfterMint" => 0,
    "Dist::Zilla::Role::AfterRelease" => 0,
    "Dist::Zilla::Role::BeforeBuild" => 0,
    "Dist::Zilla::Role::BeforeRelease" => 0,
    "Dist::Zilla::Role::Releaser" => 0,
    "Dist::Zilla::Role::TestRunner" => 0,
    "File::Spec" => 0,
    "IPC::Open3" => 0,
    "List::Util" => "1.33",
    "Moose" => 0,
    "Moose::Role" => 0,
    "String::Formatter" => "0.102082",
    "namespace::autoclean" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "Path::Class" => 0,
    "Path::Tiny" => 0,
    "Test::DZil" => 0,
    "Test::Deep" => 0,
    "Test::Fatal" => 0,
    "Test::File::ShareDir" => 0,
    "Test::More" => "0.88",
    "if" => 0
  },
  "VERSION" => "0.029",
  "test" => {
    "TESTS" => "t/*.t"
  }
);

my %FallbackPrereqs = (
  "Dist::Zilla::Role::AfterBuild" => 0,
  "Dist::Zilla::Role::AfterMint" => 0,
  "Dist::Zilla::Role::AfterRelease" => 0,
  "Dist::Zilla::Role::BeforeBuild" => 0,
  "Dist::Zilla::Role::BeforeRelease" => 0,
  "Dist::Zilla::Role::Releaser" => 0,
  "Dist::Zilla::Role::TestRunner" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "IPC::Open3" => 0,
  "List::Util" => "1.33",
  "Module::Build::Tiny" => "0.039",
  "Moose" => 0,
  "Moose::Role" => 0,
  "Path::Class" => 0,
  "Path::Tiny" => 0,
  "String::Formatter" => "0.102082",
  "Test::DZil" => 0,
  "Test::Deep" => 0,
  "Test::Fatal" => 0,
  "Test::File::ShareDir" => 0,
  "Test::More" => "0.88",
  "if" => 0,
  "namespace::autoclean" => 0,
  "strict" => 0,
  "warnings" => 0
);

unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);


Build.PL
# This Build.PL for Dist-Zilla-Plugin-Run was generated by
# Dist::Zilla::Plugin::ModuleBuildTiny::Fallback 0.014
use strict;
use warnings;

my %configure_requires = (
    'Module::Build::Tiny' => '0.039',
);

my @missing = grep {
    ! eval "require $_; $_->VERSION($configure_requires{$_}); 1"
} keys %configure_requires;

if (not @missing)
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";

    # This section for Dist-Zilla-Plugin-Run was generated by Dist::Zilla::Plugin::ModuleBuildTiny 0.008.
    use strict;
    use warnings;

    use 5.006;
    # use Module::Build::Tiny 0.039;
    Module::Build::Tiny::Build_PL();
}
else
{
    $ENV{PERL_MB_FALLBACK_SILENCE_WARNING} or warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are running this file manually, in which case you need
to learn to first fulfill all configure requires prerequisites listed in
META.yml or META.json -- or use a cpan client to install this distribution.

You can also silence this warning for future installations by setting the
PERL_MB_FALLBACK_SILENCE_WARNING environment variable, but please don't do
that until you fix your toolchain as described above.

EOW
    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));


    # This section was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.029.
    use strict;
    use warnings;

    require Module::Build; Module::Build->VERSION(0.28);


    my %module_build_args = (
      "build_requires" => {},
      "configure_requires" => {
        "Module::Build::Tiny" => "0.039",
        "perl" => "5.006"
      },
      "dist_abstract" => "Run external commands and code at specific phases of Dist::Zilla",
      "dist_author" => [
        "Torsten Raudssus <torsten\@raudss.us> L<https://raudss.us/>"
      ],
      "dist_name" => "Dist-Zilla-Plugin-Run",
      "dist_version" => "0.029",
      "license" => "perl",
      "module_name" => "Dist::Zilla::Plugin::Run",
      "recommends" => {},
      "recursive_test_files" => 1,
      "requires" => {
        "Dist::Zilla::Role::AfterBuild" => 0,
        "Dist::Zilla::Role::AfterMint" => 0,
        "Dist::Zilla::Role::AfterRelease" => 0,
        "Dist::Zilla::Role::BeforeBuild" => 0,
        "Dist::Zilla::Role::BeforeRelease" => 0,
        "Dist::Zilla::Role::Releaser" => 0,
        "Dist::Zilla::Role::TestRunner" => 0,
        "File::Spec" => 0,
        "IPC::Open3" => 0,
        "List::Util" => "1.33",
        "Moose" => 0,
        "Moose::Role" => 0,
        "String::Formatter" => "0.102082",
        "namespace::autoclean" => 0,
        "perl" => "5.006",
        "strict" => 0,
        "warnings" => 0
      },
      "script_files" => [],
      "test_requires" => {
        "ExtUtils::MakeMaker" => 0,
        "File::Spec" => 0,
        "Path::Class" => 0,
        "Path::Tiny" => 0,
        "Test::DZil" => 0,
        "Test::Deep" => 0,
        "Test::Fatal" => 0,
        "Test::File::ShareDir" => 0,
        "Test::More" => "0.88",
        "if" => 0,
        "perl" => "5.006"
      }
    );


    my %fallback_build_requires = (
      "ExtUtils::MakeMaker" => 0,
      "File::Spec" => 0,
      "Path::Class" => 0,
      "Path::Tiny" => 0,
      "Test::DZil" => 0,
      "Test::Deep" => 0,
      "Test::Fatal" => 0,
      "Test::File::ShareDir" => 0,
      "Test::More" => "0.88",
      "if" => 0,
      "perl" => "5.006"
    );


    unless ( eval { Module::Build->VERSION(0.4004) } ) {
      delete $module_build_args{test_requires};
      $module_build_args{build_requires} = \%fallback_build_requires;
    }

    my $build = Module::Build->new(%module_build_args);

    $build->create_build_script;
}
